src/lib/dt_extensions/dt_extensions.py

```diff
@@ -22,7 +25,10 @@ async def prepare_services_config_for_next_polling(current_services: List[GCPSer
             extensions_fetch_result = await extensions_fetch(gcp_session, dt_session, token)
             if not extensions_fetch_result:
                 raise Exception('Extension fetch failed')
-
+            
+            if config.metric_autodiscovery():
```

Comment:
And other thing is, I am not sure that dt_extensions should be running autodiscovery. Its like on different level.
You (lets say some main place) fetch dt_extensions, then you run autodiscovery and add to result of dt_extensions.

Reply:
So I would move this enrich one level higher (calling method)